torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -1209,7 +1209,12 @@ struct CudaGraphFuser {
 
     for (Node* node : block_->nodes()) {
       for (Block* sub_block : node->blocks()) {
-        CudaGraphFuser(sub_block, graph_).run();
+        CudaGraphFuser sub_block_cfg(sub_block, graph_);
+        sub_block_cfg.run();
+        // Accumulate runtime shapes for all sub-blocks
+        fusion_value_to_runtime_shape_.insert(
+            sub_block_cfg.fusion_value_to_runtime_shape_.begin(),
```

Comment:
This one looks promising... I am currently seeing an error `RuntimeError: fusion_value_to_runtime_size.find(self_value) != fusion_value_to_runtime_size.end()`
Hopefully it'll fix that.